### PR TITLE
Add nf-test to vcf_annotation_ensemblvep_snpeff + some general fixes

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+            [
+                
+            ],
+            [
+                "versions.yml",
+                "versions.yml",
+                "versions.yml",
+                "versions.yml",
+                "versions.yml"
+            ]
```
Comment (sateeshperi) [inline subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap:170]: blanks ?

Comment (nvnieuwk) [reply subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap:170]: Yes optional subwf outputs, I checked them and the correct ones are missing :p

File: subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf [hunk 2]
```diff
@@ -72,8 +75,8 @@ workflow VCF_ANNOTATE_ENSEMBLVEP_SNPEFF {
             .combine(ch_input.custom, by: 0) // Re-add the sample specific custom files
             .multiMap { meta, vcf, count, custom_files ->
                 // Define the new ID. The `_annotated` is to disambiguate the VEP output with its input
-                new_id = "${meta.id}${vcf.name.replace(meta.id,"").tokenize(".")[0]}_annotated" as String
-                new_meta = meta + [id:new_id]
+                def new_id = "${meta.id}${vcf.name.replace(meta.id,"").tokenize(".")[0]}_annotated" as String
+                def new_meta = meta + [id:new_id]
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf:79]:
```suggestion
```

File: subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf [hunk 3]
```diff
@@ -72,8 +75,8 @@ workflow VCF_ANNOTATE_ENSEMBLVEP_SNPEFF {
...
 
                 // Create channels: one with the VEP input and one with the original ID and count of scattered VCFs
                 input:  [ new_meta, vcf, custom_files ]
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf:82]:
```suggestion
                input:  [ meta + [id:new_id], vcf, custom_files ]
```

File: subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf [hunk 4]
```diff
@@ -83,10 +86,12 @@ workflow VCF_ANNOTATE_ENSEMBLVEP_SNPEFF {
         ch_vep_input = ch_scatter.input
     } else {
         // Use the normal input when no scattering has to be performed
-        ch_vep_input = ch_vcf.map { meta, vcf, tbi, files -> [ meta, vcf, files ] }
+        ch_vep_input = ch_vcf.map { meta, vcf, _tbi, files -> [ meta, vcf, files ] }
```
Comment (maxulysse) [inline subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf:89]: ??

Comment (jfy133) [reply subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/main.nf:88]: Agreed, why the underscore? Is this  a new notation I don't know about?!

File: subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+                        "single_end": false
+                    },
+                    "custom_test0_annotated.csv"
+                ],
+                [
+                    {
+                        "id": "custom_test1_annotated",
+                        "single_end": false
+                    },
+                    "custom_test1_annotated.csv"
```
Comment (jfy133) [inline subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap:44]: Can't even have the header string for example?

Comment (jfy133) [reply subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap:44]: Or number of rows using nft-csv?

**Review by jfy133 — APPROVED** [review]
Overall LGTM, just some minor improvement suggestions

File: subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap (hunk 5 above)
Comment (nvnieuwk) [reply subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap:44]: Good idea I forgot that plugin exists :grin: let me have a go at it

Comment (nvnieuwk) [reply subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/tests/main.nf.test.snap:44]: The CSV file isn't a valid CSV it's some sort merge of several CSV files, which causes the plugin to fail.

**Pull request merged**